Mage.Sets/src/mage/cards/k/Kitnap.java

```diff
@@ -0,0 +1,69 @@
+package mage.cards.k;
+
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.InvertCondition;
+import mage.abilities.condition.common.GiftWasPromisedCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.TapEnchantedEffect;
+import mage.abilities.effects.common.continuous.ControlEnchantedEffect;
+import mage.abilities.effects.common.counter.AddCountersAttachedEffect;
+import mage.abilities.keyword.EnchantAbility;
+import mage.abilities.keyword.GiftAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.GiftType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+/**
+ * @author TheElk801
+ */
+public final class Kitnap extends CardImpl {
+
+    private static final Condition condition = new InvertCondition(GiftWasPromisedCondition.instance);
```

Comment:
could make GiftPromisedCondition.TRUE and FALSE (or WAS, WASNT), not required though